Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -539,7 +539,29 @@ private void EnsureMixedRealityRequirements()
         /// <summary>
         /// Returns the Singleton instance of the classes type.
         /// </summary>
-        public static MixedRealityToolkit Instance => activeInstance;
+        public static MixedRealityToolkit Instance
```

Comment:
I recommend:

```
#if UNITY_EDITOR
        public static MixedRealityToolkit Instance
        {
            get
            {
                if (activeInstance != null)
                {
                    return;
                }

                var objects = FindObjectsOfType<MixedRealityToolkit>();

                switch (objects.Length)
                {
                    case 0:
                        break;

                    case 1:
                        RegisterInstance(objects[0]);
                        break;

                    default:
                        foreach (MixedRealityToolkit toolkit in objects)
                        {
                            RegisterInstance(toolkit);
                        }
                        break;
                }

                return activeInstance;
            }
        }
#else
        public static MixedRealityToolkit Instance => activeInstance;
#endif // UNITY_EDITOR
```

This is only a problem in the editor. If we only execute the search in editor, we sidestep complaints about performance.

Reply:
Great idea, this feels much safer.

Reply:
Fixed. Note that when simulating in editor, the search could technically be run, so it might slightly impact people when they are trying to profile code. However, they can always just delete the search if it's impacting their performance profiling.